framework-core/src/main/java/org/daisy/pipeline/job/impl/JobResultSetBuilder.java

```diff
@@ -48,28 +48,22 @@ static synchronized void collectOutputs(XProcScript script,XProcOutput outputs,U
 			if(prov==null)
 				continue;
 			String mediaType=script.getPortMetadata(info.getName()).getMediaType();
-			List<JobResult> results=null;
-			if(prov instanceof DynamicResultProvider){
-				results=buildJobResult((DynamicResultProvider) prov,mapper,mediaType);
-			}else{
-				results=buildJobResult(prov,mapper,mediaType);
+			if (!"application/vnd.pipeline.status+xml".equals(mediaType)) {
```

Comment:
minor comment: the media type string could be externalized somewhere as a static string…